### ISSUE TITLE
Xen 4.17.3-3 panic on boot on Framework AMD

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#9030]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

R4.2

### Brief summary

Panic on boot on Framework AMD after updating Xen to 4.17.3. 4.17.2 worked okay.

### Steps to reproduce

1. Install Qubes 4.2 on Framwork AMD
2. Add `ioapic_ack=new` to Xen options to workaround #8734 
3. Install dom0 updates.

### Expected behavior

Normal boot as 4.17.2 was.

### Actual behavior

It hits `BUG_ON` in `xen/drivers/passthrough/amd/iommu_init.c`:
```
1387 int __init amd_iommu_prepare(bool xt)
1388 {
1389     struct amd_iommu *iommu;
1390     bool has_xt = true;
1391     int rc = -ENODEV;
1392 
1393     BUG_ON( !iommu_found() );
```

<details><summary>Full Xen boot log</summary>
<p>

```
 Xen 4.17.3
(XEN) Xen version 4.17.3 (mockbuild@[unknown]) (gcc (GCC) 12.3.1 20230508 (Red Hat 12.3.1-1)) debug=n Tue Feb  6 03:30:44 GMT 2024
(XEN) Latest ChangeSet: 
(XEN) build-id: a3ce2d077c63b407f6df481100efc647c9571ad7
(XEN) Bootloader: GRUB 2.06
(XEN) Command line: placeholder console=vga,xhci loglvl=all guest_loglvl=all dbgp=xhci@pcic1:00.3,share=yes ioapic_ack=new dom0_mem=min:1024M dom0_mem=max:4096M ucode=scan smt=off gnttab_max_frames=2048 gnttab_max_maptrack_frames=4096 no-real-mode edd=off
(XEN) Xen image load base address: 0x4f400000
(XEN) Video information:
(XEN)  VGA is graphics mode 1024x768, 32 bpp
(XEN) Disc information:
(XEN)  Found 0 MBR signatures
(XEN)  Found 1 EDD information structures
(XEN) CPU Vendor: AMD, Family 25 (0x19), Model 116 (0x74), Stepping 1 (raw 00a70f41)
(XEN) Enabling Supervisor Shadow Stacks
(XEN) EFI RAM map:
(XEN)  [0000000000000000, 000000000009efff] (usable)
(XEN)  [000000000009f000, 00000000000bffff] (reserved)
(XEN)  [0000000000100000, 0000000009afffff] (usable)
(XEN)  [0000000009b00000, 0000000009dfffff] (reserved)
(XEN)  [0000000009e00000, 0000000009efffff] (usable)
(XEN)  [0000000009f00000, 0000000009f3bfff] (ACPI NVS)
(XEN)  [0000000009f3c000, 000000004235ffff] (usable)
(XEN)  [0000000042360000, 000000004455ffff] (reserved)
(XEN)  [0000000044560000, 0000000044568fff] (usable)
(XEN)  [0000000044569000, 000000004456cfff] (reserved)
(XEN)  [000000004456d000, 000000004456efff] (usable)
(XEN)  [000000004456f000, 000000004456ffff] (reserved)
(XEN)  [0000000044570000, 000000005077efff] (usable)
(XEN)  [000000005077f000, 0000000052f7efff] (reserved)
(XEN)  [0000000052f7f000, 000000005af7efff] (ACPI NVS)
(XEN)  [000000005af7f000, 000000005affefff] (ACPI data)
(XEN)  [000000005afff000, 000000005affffff] (usable)
(XEN)  [000000005b000000, 000000005bffffff] (reserved)
(XEN)  [000000005d790000, 000000005d7effff] (reserved)
(XEN)  [000000005d7f5000, 000000005fffffff] (reserved)
(XEN)  [00000000c0300000, 00000000c03fffff] (reserved)
(XEN)  [00000000fed80000, 00000000fed80fff] (reserved)
(XEN)  [00000000ff000000, 00000000ffffffff] (reserved)
(XEN)  [0000000100000000, 000000087e2fffff] (usable)
(XEN)  [0000000880000000, 00000008c01fffff] (reserved)
(XEN) ACPI: RSDP 5AFFE014, 0024 (r2 INSYDE)
(XEN) ACPI: XSDT 5AFA0228, 017C (r1 INSYDE EDK2            2       1000013)
(XEN) ACPI: FACP 5AFEE000, 010C (r5 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI Warning (tbfadt-0459): Optional field "Pm2ControlBlock" has zero address or length: 0000000000000000/1 [20070126]
(XEN) ACPI: DSDT 5AFDE000, 93A6 (r2 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: FACS 5AED4000, 0040
(XEN) ACPI: UEFI 5AF67000, 01CF (r1 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFF4000, 8416 (r2 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: SSDT 5AFF3000, 033E (r1 INSYDE EDK2         1000 ACPI    40000)
(XEN) ACPI: SSDT 5AFF1000, 0ABD (r2 INSYDE EDK2         1000 ACPI    40000)
(XEN) ACPI: ASF! 5AFF0000, 00A5 (r32 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: BOOT 5AFEF000, 0028 (r1 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: HPET 5AFED000, 0038 (r1 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: APIC 5AFEC000, 0138 (r3 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: MCFG 5AFEB000, 003C (r1 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: SLIC 5AFEA000, 0176 (r1 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: VFCT 5AFD9000, 4284 (r1 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFD8000, 00F8 (r2 INSYDE EDK2         1000 ACPI    40000)
(XEN) ACPI: SSDT 5AFD4000, 3F6A (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: CRAT 5AFD3000, 0B90 (r1 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: CDIT 5AFD2000, 0029 (r1 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFD1000, 0F8D (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFD0000, 0EC1 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFFD000, 0931 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFCD000, 13EC (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFCC000, 077A (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFCB000, 0737 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFC9000, 15C8 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFC6000, 2A8F (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFBC000, 9821 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: FPDT 5AFBB000, 0044 (r1 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: BGRT 5AFBA000, 0038 (r1 INSYDE EDK2            2 ACPI    40000)
(XEN) ACPI: WSMT 5AFB9000, 0028 (r1 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: TPM2 5AFE9000, 0050 (r5 INSYDE EDK2     20505348 ACPI    40000)
(XEN) ACPI: MHSP 5AFE8000, 00C8 (r4 INSYDE EDK2     20505348 ACPI    40000)
(XEN) ACPI: SSDT 5AFB8000, 00E5 (r2 INSYDE EDK2            4 ACPI    40000)
(XEN) ACPI: IVRS 5AFB7000, 01A4 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFB6000, 0747 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFB5000, 0C88 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFB4000, 0057 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFB2000, 170B (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFB1000, 0FF7 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFA7000, 97E3 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFA2000, 4FEB (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFA1000, 0C7F (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AFCF000, 0956 (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AF9F000, 008D (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) ACPI: SSDT 5AF9E000, 0EAD (r2 INSYDE EDK2            1 ACPI    40000)
(XEN) System RAM: 31940MB (32707448kB)
(XEN) No NUMA configuration found
(XEN) Faking a node at 0000000000000000-000000087e300000
(XEN) Domain heap initialised
(XEN) vesafb: framebuffer at 0x0000008000000000, mapped to 0xffff82c000201000, using 3072k, total 3072k
(XEN) vesafb: mode is 1024x768x32, linelength=4096, font 8x14
(XEN) vesafb: Truecolor: size=8:8:8:8, shift=24:16:8:0
(XEN) SMBIOS 3.5 present.
(XEN) Using APIC driver default
(XEN) ACPI: PM-Timer IO Port: 0x408 (32 bits)
(XEN) ACPI: v5 SLEEP INFO: control[0:0], status[0:0]
(XEN) ACPI: SLEEP INFO: pm1x_cnt[1:404,1:0], pm1x_evt[1:400,1:0]
(XEN) ACPI: 32/64X FACS address mismatch in FADT - 5aed4000/0000000000000000, using 32
(XEN) ACPI:             wakeup_vec[5aed400c], vec_size[20]
(XEN) ACPI: Local APIC address 0xfee00000
(XEN) Overriding APIC driver with bigsmp
(XEN) ACPI: IOAPIC (id[0x21] address[0xfec00000] gsi_base[0])
(XEN) IOAPIC[0]: apic_id 33, version 33, address 0xfec00000, GSI 0-23
(XEN) ACPI: IOAPIC (id[0x22] address[0xfec01000] gsi_base[24])
(XEN) IOAPIC[1]: apic_id 34, version 33, address 0xfec01000, GSI 24-55
(XEN) ACPI: INT_SRC_OVR (bus 0 bus_irq 0 global_irq 2 dfl dfl)
(XEN) ACPI: INT_SRC_OVR (bus 0 bus_irq 9 global_irq 9 low level)
(XEN) ACPI: IRQ0 used by override.
(XEN) ACPI: IRQ2 used by override.
(XEN) ACPI: IRQ9 used by override.
(XEN) ACPI: HPET id: 0x10228210 base: 0xfed00000
(XEN) PCI: MCFG configuration 0: base e0000000 segment 0000 buses 00 - ff
(XEN) PCI: Not using MCFG for segment 0000 bus 00-ff
(XEN) ACPI: BGRT: invalidating v1 image at 0x4b3ab018
(XEN) Using ACPI (MADT) for SMP configuration information
(XEN) SMP: Allowing 16 CPUs (4 hotplug CPUs)
(XEN) IRQ limits: 56 GSI, 2440 MSI/MSI-X
(XEN) AMD-Vi: IOMMU Extended Features:
(XEN) - Peripheral Page Service Request
(XEN) - NX bit
(XEN) - Guest APIC Physical Processor Interrupt
(XEN) - Invalidate All Command
(XEN) - Guest APIC
(XEN) - Performance Counters
(XEN) - Host Address Translation Size: 0x2
(XEN) - Guest Address Translation Size: 0
(XEN) - Guest CR3 Root Table Level: 0x1
(XEN) - Maximum PASID: 0xf
(XEN) - SMI Filter Register: 0x1
(XEN) - SMI Filter Register Count: 0x1
(XEN) - Guest Virtual APIC Modes: 0x1
(XEN) - Dual PPR Log: 0x2
(XEN) - Dual Event Log: 0x2
(XEN) - Secure ATS
(XEN) - User / Supervisor Page Protection
(XEN) - Device Table Segmentation: 0x3
(XEN) - PPR Log Overflow Early Warning
(XEN) - PPR Automatic Response
(XEN) - Memory Access Routing and Control: 0x1
(XEN) - Block StopMark Message
(XEN) - Performance Optimization
(XEN) - MSI Capability MMIO Access
(XEN) - Guest I/O Protection
(XEN) - Enhanced PPR Handling
(XEN) - Invalidate IOTLB Type
(XEN) - VM Table Size: 0x2
(XEN) - Guest Access Bit Update Disable
(XEN) AMD-Vi: Warning: IVMD: [4f83f000,4f880000) is not (entirely) in reserved memory
(XEN) AMD-Vi: Error: IVMD: page at 4f83f000 can't be converted
(XEN) Switched to APIC driver x2apic_mixed
(XEN) CPU0: 1600 ... 3500 MHz
(XEN) xstate: size: 0x988 and states: 0x2e7
(XEN) CPU0: AMD Fam19h machine check reporting enabled
(XEN) Vulnerable to SRSO, without suitable microcode to mitigate
(XEN) Speculative mitigation facilities:
(XEN)   Hardware hints: STIBP_ALWAYS IBRS_FAST IBRS_SAME_MODE BTC_NO IBPB_RET
(XEN)   Hardware features: IBPB IBRS STIBP SSBD PSFD L1D_FLUSH
(XEN)   Compiled-in support: INDIRECT_THUNK
(XEN)   Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ STIBP+ SSBD- PSFD-, Other: IBPB-ctxt BRANCH_HARDEN
(XEN)   Support for HVM VMs: MSR_SPEC_CTRL MSR_VIRT_SPEC_CTRL RSB
(XEN)   Support for PV VMs: None
(XEN)   XPTI (64-bit PV only): Dom0 disabled, DomU disabled (without PCID)
(XEN)   PV L1TF shadowing: Dom0 disabled, DomU disabled
(XEN) Using scheduler: SMP Credit Scheduler rev2 (credit2)
(XEN) Initializing Credit2 scheduler
(XEN)  load_precision_shift: 18
(XEN)  load_window_shift: 30
(XEN)  underload_balance_tolerance: 0
(XEN)  overload_balance_tolerance: -3
(XEN)  runqueues arrangement: socket
(XEN)  cap enforcement granularity: 10ms
(XEN) load tracking window length 1073741824 ns
(XEN) Platform timer is 14.318MHz HPET
(XEN) Detected 3493.417 MHz processor.
(XEN) Freed 1020kB unused BSS memory
(XEN) EFI memory map:
(XEN)  0000000000000-0000000003fff type=2 attr=000000000000000f
(XEN)  0000000004000-0000000086fff type=7 attr=000000000000000f
(XEN)  0000000087000-0000000087fff type=4 attr=000000000000000f
(XEN)  0000000088000-000000008efff type=7 attr=000000000000000f
(XEN)  000000008f000-000000009efff type=2 attr=000000000000000f
(XEN)  000000009f000-000000009ffff type=0 attr=000000000000000f
(XEN)  0000000100000-0000004985fff type=2 attr=000000000000000f
(XEN)  0000004986000-0000009afffff type=7 attr=000000000000000f
(XEN)  0000009b00000-0000009dfffff type=0 attr=000000000000000f
(XEN)  0000009e00000-0000009efffff type=7 attr=000000000000000f
(XEN)  0000009f00000-0000009f3bfff type=10 attr=000000000000000f
(XEN)  0000009f3c000-00000226cffff type=7 attr=000000000000000f
(XEN)  00000226d0000-000003437efff type=1 attr=000000000000000f
(XEN)  000003437f000-00000343fefff type=4 attr=000000000000000f
(XEN)  00000343ff000-000003fd9afff type=7 attr=000000000000000f
(XEN)  000003fd9b000-000003fffdfff type=1 attr=000000000000000f
(XEN)  000003fffe000-00000403fffff type=4 attr=000000000000000f
(XEN)  0000040400000-000004049ffff type=7 attr=000000000000000f
(XEN)  00000404a0000-00000404bffff type=4 attr=000000000000000f
(XEN)  00000404c0000-00000404d6fff type=3 attr=000000000000000f
(XEN)  00000404d7000-00000404e7fff type=4 attr=000000000000000f
(XEN)  00000404e8000-0000040518fff type=3 attr=000000000000000f
(XEN)  0000040519000-000004056bfff type=4 attr=000000000000000f
(XEN)  000004056c000-0000040574fff type=3 attr=000000000000000f
(XEN)  0000040575000-0000041d42fff type=4 attr=000000000000000f
(XEN)  0000041d43000-0000041d5ffff type=3 attr=000000000000000f
(XEN)  0000041d60000-000004235ffff type=4 attr=000000000000000f
(XEN)  0000042360000-000004455ffff type=0 attr=000000000000000f
(XEN)  0000044560000-0000044566fff type=3 attr=000000000000000f
(XEN)  0000044567000-0000044568fff type=4 attr=000000000000000f
(XEN)  0000044569000-000004456cfff type=6 attr=800000000000000f
(XEN)  000004456d000-000004456efff type=4 attr=000000000000000f
(XEN)  000004456f000-000004456ffff type=6 attr=800000000000000f
(XEN)  0000044570000-0000044577fff type=7 attr=000000000000000f
(XEN)  0000044578000-0000044578fff type=2 attr=000000000000000f
(XEN)  0000044579000-000004abd0fff type=7 attr=000000000000000f
(XEN)  000004abd1000-000004abd1fff type=4 attr=000000000000000f
(XEN)  000004abd2000-000004abd3fff type=7 attr=000000000000000f
(XEN)  000004abd4000-000004e77efff type=4 attr=000000000000000f
(XEN)  000004e77f000-000004f5fffff type=7 attr=000000000000000f
(XEN)  000004f600000-000004f9b3fff type=2 attr=000000000000000f
(XEN)  000004f9b4000-000004f9f9fff type=7 attr=000000000000000f
(XEN)  000004f9fa000-000005077efff type=3 attr=000000000000000f
(XEN)  000005077f000-0000050f7efff type=5 attr=800000000000000f
(XEN)  0000050f7f000-0000051f7efff type=6 attr=800000000000000f
(XEN)  0000051f7f000-0000052f7efff type=0 attr=000000000000000f
(XEN)  0000052f7f000-000005af7efff type=10 attr=000000000000000f
(XEN)  000005af7f000-000005affefff type=9 attr=000000000000000f
(XEN)  000005afff000-000005affffff type=4 attr=000000000000000f
(XEN)  0000100000000-000087e2fffff type=7 attr=000000000000000f
(XEN)  00000000a0000-00000000bffff type=0 attr=0000000000000000
(XEN) Unknown cachability for MFNs 0xa0-0xbf
(XEN)  000005b000000-000005bffffff type=0 attr=0000000000000009
(XEN)  000005d790000-000005d7effff type=0 attr=0000000000000000
(XEN) Unknown cachability for MFNs 0x5d790-0x5d7ef
(XEN)  000005d7f5000-000005fffffff type=0 attr=0000000000000000
(XEN) Unknown cachability for MFNs 0x5d7f5-0x5ffff
(XEN)  00000c0300000-00000c03fffff type=11 attr=8000000000000001
(XEN)  00000fed80000-00000fed80fff type=11 attr=8000000000000001
(XEN)  00000ff000000-00000ffffffff type=11 attr=8000000000000001
(XEN)  0000880000000-000089fffffff type=0 attr=0000000000000000
(XEN) Unknown cachability for MFNs 0x880000-0x89ffff
(XEN)  00008a0000000-00008c01fffff type=11 attr=8000000000000001
(XEN) alt table ffff82d04041c390 -> ffff82d0404270a6
(XEN) Xen BUG at drivers/passthrough/amd/iommu_init.c:1386
(XEN) ----[ Xen-4.17.3  x86_64  debug=n  Not tainted ]----
(XEN) CPU:    0
(XEN) RIP:    e008:[<ffff82d0403a5ef8>] amd_iommu_prepare+0x288/0x390
(XEN) RFLAGS: 0000000000010246   CONTEXT: hypervisor
(XEN) rax: ffff82d0403a6620   rbx: ffff82d04042c7d0   rcx: 0000000000000000
(XEN) rdx: ffff82d0403e8510   rsi: ffff82d0403e0780   rdi: 0000000000000000
(XEN) rbp: 0000000000000000   rsp: ffff82d0403dfd28   r8:  0000000000000010
(XEN) r9:  0000000000000000   r10: 7fffffffffffffff   r11: 0000000000000000
(XEN) r12: ffff82d040430b20   r13: ffff82d0403dfe48   r14: ffff82d0403814d8
(XEN) r15: ffff83000009de80   cr0: 000000008005003b   cr4: 0000000000f506e0
(XEN) cr3: 000000004f83b000   cr2: 0000000000000000
(XEN) fsb: 0000000000000000   gsb: 0000000000000000   gss: 0000000000000000
(XEN) ds: 0000   es: 0000   fs: 0000   gs: 0000   ss: 0000   cs: e008
(XEN) Xen code around <ffff82d0403a5ef8> (amd_iommu_prepare+0x288/0x390):
(XEN)  fd ff ff 0f 1f 44 00 00 <0f> 0b 66 0f 1f 44 00 00 0f 0b 80 3d d7 68 08 00
(XEN) Xen stack trace from rsp=ffff82d0403dfd28:
(XEN)    0000000000000100 0000000000000000 0000000000000000 ffff82d040430b20
(XEN)    ffff82d0403dfe48 ffff82d0403814d8 ffff83000009de80 ffff82d0403a6019
(XEN)    ffff82d0403dffff 0000000000000000 ffff82d0403dffff 0000000000000000
(XEN)    ffff82d040430b20 ffff82d0403dfe48 ffff82d0403814d8 ffff83000009de80
(XEN)    ffff82d0403a6646 ffff82d0403aa3fc ffff82d0403dffff 0000000000000000
(XEN)    ffff82d0403cb802 ffff83000009dfb0 ffff83000009de80 ffff83000000000c
(XEN)    0000000000000000 ffff8308750a8000 00000000047f9000 000000000087e300
(XEN)    ffff82d0403dfef8 ffff83000009df80 00000000000001ff 000000087e300000
(XEN)    000000000009df80 ffff83000009de80 0000000000000000 0000000000000000
(XEN)    0000000000000002 ffff82d040481000 ffff82d040580000 0000000800000000
(XEN)    000000010000006e 0000000000000003 00000000000002f8 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN)    00000000404bf240 0000000000000000 00000000404bfbb8 00000000404bfbc0
(XEN)    000000004f7af4d1 ffff82d040273734 0000000000000000 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN)    0000000000000000 0000000000000000 0000000000000000 0000000000000000
(XEN) Xen call trace:
(XEN)    [<ffff82d0403a5ef8>] R amd_iommu_prepare+0x288/0x390
(XEN)    [<ffff82d0403a6019>] S amd_iommu_init+0x19/0x3c0
(XEN)    [<ffff82d0403a6646>] S pci_amd_iommu.c#iov_detect+0x26/0x70
(XEN)    [<ffff82d0403aa3fc>] S iommu_setup+0x3c/0x230
(XEN)    [<ffff82d0403cb802>] S __start_xen+0x1672/0x264b
(XEN)    [<ffff82d040273734>] S __high_start+0x94/0xa0
(XEN) 
(XEN) 
(XEN) ****************************************
(XEN) Panic on CPU 0:
(XEN) Xen BUG at drivers/passthrough/amd/iommu_init.c:1386
(XEN) ****************************************
(XEN) 
(XEN) Reboot in five seconds...
```

</p>
</details> 

https://github.com/QubesOS/qubes-vmm-xen/pull/179 likely is related
There are also two possibly related patch series:
- https://lore.kernel.org/xen-devel/20240201170159.66330-1-roger.pau@citrix.com/
- https://lore.kernel.org/xen-devel/20240214103741.16189-1-roger.pau@citrix.com/

**Comment by marmarek** [comment]
This seems to affect only using XHCI debug console.

**Comment by juju4** [comment]
Can you confirm that 4.2.3 iso installer has the patched xen? It's Xen-4.17.4, no information on patchset. I still observed this issue as said in https://forum.qubes-os.org/t/qubes-hcl-azw-ser8-beelink-ser8/30493
Thanks
